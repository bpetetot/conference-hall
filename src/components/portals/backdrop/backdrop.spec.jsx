/* eslint-env jest */
import React from 'react'
import snap from 'tests/snapshot'
import { shallow } from 'enzyme'
import Backdrop from './index.js'

const snapshot = (props) => snap(Backdrop)({ ...props })

describe('components/backdrop', () => {
  it('should render', snapshot({}))
  it('should call onClick by default', () => {
    const onclick = jest.fn()
    const wrapper = shallow(<Backdrop onClick={onclick} />)
    wrapper.find('.backdrop').simulate('click')
    expect(onclick).toHaveBeenCalled()
  })
  it('should not call onClick if not ', () => {
    const onclick = jest.fn()
    const wrapper = shallow(<Backdrop onClick={onclick} withClickOutside={false} />)
    wrapper.find('.backdrop').simulate('click')
    expect(onclick).not.toHaveBeenCalled()
  })
})
