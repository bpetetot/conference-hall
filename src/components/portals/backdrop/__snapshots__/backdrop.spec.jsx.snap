// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/backdrop should render 1`] = `
<div
  className="backdrop"
  role="button"
/>
`;
